docs/spam_checker.rst

```diff
@@ -0,0 +1,85 @@
+# Handling spam in Synapse
```

Comment:
generally new docs get written in markdown rather than rst I think... I'm not that bothered about it though.

Reply:
Converted! This doc was written as a weird mixture of the two anyway.